### ISSUE TITLE
DevTools reports 100% CPU while doing nothing

### ISSUE THREAD

**Issue by alexgleason** [denoland/deno#21620]
Version: Deno 1.39.0  
Chromium: 120.0.6099.71

I created a simple reproduction script, `gg.ts`:

```ts
while (true) {
  await new Promise((resolve) => setTimeout(resolve, 1000));
}
```

Running it with `deno run --inspect gg.ts` and then viewing it in Chromium DevTools, the histogram at the top shows 100% CPU usage. 

![image](https://github.com/denoland/deno/assets/3639540/e0e0cd16-313c-4b60-b1ef-4be86ca392a4)

You can check my results in your DevTools here: [CPU-20231217T142344.cpuprofile](https://github.com/denoland/deno/files/13697665/CPU-20231217T142344.cpuprofile)

This started happening recently, and I don't know why. The CPU itself is not close to 100% on any core when viewed with `htop`. The problem does not occur when profiling a normal webpage in the same browser, just when inspecting Deno.

**Comment by bartlomieju** [comment]
I believe this diagram doesn't show actual CPU usage, but show percentage of time the program spents in CPU. Timers are known to have poor performance: https://github.com/denoland/deno/issues/20356 and there's ongoing work to speed them up: https://github.com/denoland/deno/pull/20428

**Comment by alexgleason** [comment]
I was able to dig up an old screenshot from some Deno profiling I did on November 20th:

![Screenshot from 2023-11-20 19-21-32](https://github.com/denoland/deno/assets/3639540/7af2b7bb-9ddd-4ab5-bcae-eb46b4d417d6)

The CPU chart at the top looks correct in this version.

I tried reverting my codebase to an old version (and downgrading to Deno 1.36), and I'm still having the problem in this issue. So that suggests to me something changed in Chromium rather than Deno, that makes Deno profiling a lot less useful.

**Comment by bartlomieju** [comment]
> I tried reverting my codebase to an old version (and downgrading to Deno 1.36), and I'm still having the problem in this issue. So that suggests to me something changed in Chromium rather than Deno, that makes Deno profiling a lot less useful.

I agree, IIRC a couple months back `(program)` was colored in gray and didn't count towards "Scripting" category (which it does now). 

Chrome DevTools devs asked for feedback on this panel, which appears to just been closed: https://github.com/ChromeDevTools/rfcs/discussions/2 I suggest to open an issue with them. I'll be happy to coordinate on that.

**Comment by alexgleason** [comment]
Thanks @bartlomieju, opened an issue here: https://bugs.chromium.org/p/chromium/issues/detail?id=1512463

**Comment by june07** [comment]
See https://github.com/denoland/deno/issues/24877#issuecomment-2276989796 for a workaround/fix that is just as applicable to this issue.

**Comment by alexgleason** [comment]
This issue is still occurring on Deno 2.0 and Chromium 129 (the latest version). The people on the Chromium issue tracker didn't want to install Deno to test it. :joy:
